Hca errorable select

akainic opened the pull request:
ErrorableSelect component with tests. Also included file for arrays of options for selects. 

ErrorableSelect used so far for: 
1. Marital status
2. State (in veteran address)
3. Country (in veteran address)

Still to do: 
1. Figure out why logic from @awong-dev's version for `options` prop is not working
2. `rake lint` not working 
3. Console errors related to VaMedicalFacility, no idea why

akainic committed:
WIP create errorable select

awong-dev commented on _health-care/_js/components/form-elements/ErrorableSelect.jsx:
```diff
@@ -0,0 +1,107 @@
+      );
+    });
+    // let reactKey = 0;
+    // const optionElements = this.props.options.map((obj) => {
```
The reason yours and mine diverges is I expect the options prop to be an array of objects where each object has a `label` and a `value` property.

In your API, you are expecting an associative array.

awong-dev committed:
Make ErrorableSelect options accept array.

Previously the ErrorableSelect option prop took a JS object which
was used to map the (key, value) pairs into (value, label) on an
<option> element.

This has the undesirable effect of requiring preventing values from
having spaces or being JS keywords.

Change the API so that options is an array of either strings, or objects
with an explicit {label:, value:} structure.

akainic commented:
Merging this, TBR in more detail later
